### PR TITLE
fix: get job run from Optimus DB

### PR DESCRIPTION
**Pull request by Mryashbhardwaj**

### CODE REVIEW

File: core/scheduler/handler/v1beta1/job_run.go [hunk 1]
```diff
@@ -80,8 +82,51 @@ func (h JobRunHandler) JobRunInput(ctx context.Context, req *pb.JobRunInputReque
 	}, nil
 }
 
-// JobRun currently gets the job runs from scheduler based on the criteria
-// TODO: later should collect the job runs from optimus
+// GetJobRun gets job runs from optimus DB based on the criteria
+func (h JobRunHandler) GetJobRun(ctx context.Context, req *pb.GetJobRunsRequest) (*pb.GetJobRunsResponse, error) {
```
Comment (deryrahman) [inline core/scheduler/handler/v1beta1/job_run.go:86]: can we provide test for this handler?

File: core/scheduler/service/job_run_service.go [hunk 2]
```diff
@@ -146,6 +149,27 @@ func (s *JobRunService) JobRunInput(ctx context.Context, projectName tenant.Proj
 	return s.compiler.Compile(ctx, details, config, executedAt)
 }
 
+func (s *JobRunService) GetJobRunsByFilter(ctx context.Context, projectName tenant.ProjectName, jobName scheduler.JobName, filters ...filter.FilterOpt) ([]*scheduler.JobRun, error) {
```
Comment (deryrahman) [inline core/scheduler/service/job_run_service.go:152]: please also add test for this function as well

File: internal/store/postgres/scheduler/job_run_repository.go [hunk 3]
```diff
@@ -90,6 +91,56 @@ func (j *JobRunRepository) GetByID(ctx context.Context, id scheduler.JobRunID) (
 	return jr.toJobRun()
 }
 
+func (j *JobRunRepository) GetLatestRun(ctx context.Context, project tenant.ProjectName, jobName scheduler.JobName, runState *scheduler.State) (*scheduler.JobRun, error) {
```
Comment (deryrahman) [inline internal/store/postgres/scheduler/job_run_repository.go:94]: please also provide test `GetLatestRun` and `GetRunsByTimeRange`

**Review by deryrahman — CHANGES_REQUESTED** [review]

**Comment by deryrahman** [comment]
can you update proto commit reference in Makefile? I believe this should be changed according to this [PR](https://github.com/goto/proton/pull/129)

File: core/scheduler/handler/v1beta1/job_run.go (hunk 1 above)
Comment (Mryashbhardwaj) [reply core/scheduler/handler/v1beta1/job_run.go:86]: sure

File: core/scheduler/service/job_run_service.go (hunk 2 above)
Comment (Mryashbhardwaj) [reply core/scheduler/service/job_run_service.go:152]: sure

**Comment by Mryashbhardwaj** [comment]
> can you update proto commit reference in Makefile? I believe this should be changed according to this [PR](https://github.com/goto/proton/pull/129)

updated the proton commit

**Comment by deryrahman** [comment]
LGTM

**Review by deryrahman — APPROVED** [review]

**Pull request merged**